web/hooks/use-query-params.ts

```diff
@@ -0,0 +1,210 @@
+'use client'
+
+/**
+ * Centralized URL query parameter management hooks using nuqs
+ *
+ * This file provides type-safe, performant query parameter management
+ * that doesn't trigger full page refreshes (shallow routing).
+ *
+ * Best practices from nuqs documentation:
+ * - Use useQueryState for single parameters
+ * - Use useQueryStates for multiple related parameters (atomic updates)
+ * - Always provide parsers with defaults for type safety
+ * - Use shallow routing to avoid unnecessary re-renders
+ */
+
+import type { Options } from 'nuqs'
+import {
+
+  parseAsArrayOf,
+  parseAsString,
+  useQueryState,
+  useQueryStates,
+} from 'nuqs'
+import { ACCOUNT_SETTING_MODAL_ACTION } from '@/app/components/header/account-setting/constants'
+
+/**
+ * Modal State Query Parameters
+ * Manages modal visibility and configuration via URL
+ */
+export const PRICING_MODAL_QUERY_PARAM = 'pricing'
+export const PRICING_MODAL_QUERY_VALUE = 'open'
+
+/**
+ * Hook to manage pricing modal state via URL
+ * @returns [isOpen, setIsOpen] - Tuple like useState
+ *
+ * @example
+ * const [isPricingModalOpen, setIsPricingModalOpen] = usePricingModal()
+ * setIsPricingModalOpen(true) // Sets ?pricing=open
+ * setIsPricingModalOpen(false) // Removes ?pricing
+ */
+export function usePricingModal() {
+  const [isOpen, setIsOpen] = useQueryState(
+    PRICING_MODAL_QUERY_PARAM,
+    parseAsString,
+  )
+
+  const setIsOpenBoolean = (open: boolean, options?: Options) => {
+    const history = options?.history ?? (open ? 'push' : 'replace')
+    setIsOpen(open ? PRICING_MODAL_QUERY_VALUE : null, { ...options, history })
+  }
+
+  return [isOpen === PRICING_MODAL_QUERY_VALUE, setIsOpenBoolean] as const
+}
```

Comment:
We can use a custom parser for this if you want to keep the args as before.

https://nuqs.dev/docs/parsers/making-your-own